src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -431,6 +431,22 @@ public static SymbolInfo GetSymbolInfo(this SemanticModel semanticModel, SelectO
         }
 
         /// <summary>
+        /// Gets the symbol information for the property of a sub-property pattern.
+        /// </summary>
+        public static SymbolInfo GetSymbolInfo(this SemanticModel semanticModel, SubPropertyPatternSyntax node, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
I don't see a similarity, ConstructorInitializerSyntax actually represents a constructor reference. Also, I believe SymbolInfo cannot be requested for a token and base/this are tokens. So, there is only one node that can be used to get information about the constructor used. Similar with an attribute, it represents a constructor reference.
I would say that SubPropertyPatternSyntax node is somewhat similar to a member initialization withing object initializer, which is represented by an assignment expression node. I would be surprised if GetSymbolInfo returned the target property given the assignment node. One actually have to pass the Left to get this information.